Source/WebGPU/WGSL/ConstantValue.h

```diff
@@ -148,6 +148,10 @@ struct ConstantValue : BaseValue {
             return *u32;
         if (auto* abstractInt = std::get_if<int64_t>(this))
             return *abstractInt;
+        if (auto* abstractFloat = std::get_if<float>(this)) {
+            RELEASE_ASSERT(*abstractFloat == 0.f);
```

Comment:
@tadeuzagallo Should this follow the C rules of converting a float to int by truncation?

The spec (https://www.w3.org/TR/WGSL/#conversion-rank) appears to ban the automatic conversion of float to int.

Reply:
I don't think this is correct. It seems like we have a mismatch between the constant and the type, which is incorrect. We're probably not converting the override initial value, which we need to.

Reply:
I agree, I will investigate a bit further to see what is occurring here.

Reply:
@djg @tadeuzagallo ah so we were using mangledName where we should not have been. 

PR is updated when you have a minute